src/App.tsx

```diff
@@ -1,14 +1,83 @@
 /* eslint-disable max-len */
-import React from 'react';
+import { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [todoSelector, setTodoSelector] = useState('all');
+  const [todoFilter, setTodoFilter] = useState('');
+
+  const setInitialTodos = async () => {
+    const data = await getTodos();
+
+    setTodos(data);
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    setInitialTodos();
+  }, []);
+
+  const getVisibleTodos = () => {
+    const needsToFilter
+      = todoSelector === 'active' || todoSelector === 'completed';
+
+    if (!needsToFilter && !todoFilter) {
+      return todos;
+    }
+
+    return todos.filter((todo) => {
+      const todoTitleIncludesFilter = todo.title
+        .toLowerCase()
+        .includes(todoFilter.toLowerCase());
+
+      switch (todoSelector) {
+        case 'active':
+          return !todo.completed && todoTitleIncludesFilter;
+        case 'completed':
+          return todo.completed && todoTitleIncludesFilter;
+        default:
+          return todoTitleIncludesFilter;
+      }
+    });
+  };
+
+  const visibleTodos = useMemo(getVisibleTodos, [
+    todos,
+    todoSelector,
+    todoFilter,
+  ]);
+
+  const showTodoDetails = (todo: Todo | null) => () => {
+    setSelectedTodo(todo);
+  };
+
+  const hideTodoDetails = () => {
```

Comment:
Consider renaming 'hideTodoDetails' to 'handleHideTodoDetails' for better readability and consistency with other function names.